File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/dialog/DemoBaseDialog.java
package io.ganguo.demo.view.dialog;

import android.content.Context;

import io.ganguo.demo.databinding.ItemDemoBinding;
import io.ganguo.demo.viewmodel.item.ItemDemoVModel;
import io.ganguo.utils.common.ToastHelper;
import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.vmodel.ViewModelDialog;
import io.ganguo.utils.util.Randoms;

/**
 * Created by Roger on 7/13/16.
 */
public class DemoBaseDialog extends ViewModelDialog<ItemDemoBinding, ItemDemoVModel> {

    public DemoBaseDialog(Context context) {
        super(context);
    }

    @Override
    public ItemDemoVModel<String> createViewModel() {
        return createItemDemoVModel();
    }

    @Override
    public void onViewAttached(ItemDemoVModel viewModel) {

    }


    /**
     * function: create Item ViewModel
     *
     * @return
     */
    public ItemDemoVModel<String> createItemDemoVModel() {
        String content = Randoms.getRandomCapitalLetters(12);
        return new ItemDemoVModel<String>()
                .setBtnText("click")
                .setData(content)
                .setContent(content)
                .setClickAction(s -> dismiss());
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/dialog/DemoBaseDialog.kt
package io.ganguo.demo.view.dialog

import android.content.Context
import android.view.ViewGroup

import io.ganguo.demo.databinding.ItemDemoBinding
import io.ganguo.demo.viewmodel.item.ItemDemoVModel
import io.ganguo.viewmodel.core.view.ViewModelDialog
import io.ganguo.utils.util.Randoms

/**
 * Created by Roger on 7/13/16.
 */
class DemoBaseDialog(context: Context) : ViewModelDialog<ItemDemoBinding, ItemDemoVModel<*>>(context) {

    override val dialogWidth: Int
        get() = ViewGroup.LayoutParams.MATCH_PARENT
    override val dialogHeight: Int
        get() = ViewGroup.LayoutParams.MATCH_PARENT

    override fun createViewModel(): ItemDemoVModel<String> {
        return createItemDemoVModel()
    }



    /**
     * function: create Item ViewModel
     *
     * @return
     */
    fun createItemDemoVModel(): ItemDemoVModel<String> {
        val content = Randoms.getRandomCapitalLetters(12)
        return ItemDemoVModel<String>()
                .setBtnText("click")
                .setDataObj(content)
                .setContent(content)
                .setClickAction {
                    dismiss()
                }
    }

    override fun onViewAttached(viewModel: ItemDemoVModel<*>) {
    }

}
